core/server/src/test/java/alluxio/worker/file/UnderFileSystemManagerTest.java

```diff
@@ -121,4 +129,114 @@ public void cancelNonExistentUfsFileTest() throws Exception {
+   * exists.
+   */
+  @Test
+  public void openUfsFileTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(true);
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    manager.openFile(new AlluxioURI(uniqPath));
+    Mockito.verify(mMockUfs).exists(uniqPath);
+  }
+
+  /**
+   * Tests opening a file fails when the file does not exist.
+   */
+  @Test
+  public void openNonExistentUfsFileTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(false);
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    mThrown.expect(FileDoesNotExistException.class);
+    manager.openFile(new AlluxioURI(uniqPath));
+  }
+
+  /**
+   * Tests closing an opened file invalidates the id.
+   */
+  @Test
+  public void closeUfsFileTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(true);
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    long id = manager.openFile(new AlluxioURI(uniqPath));
+    manager.closeFile(id);
+    mThrown.expect(FileDoesNotExistException.class);
+    manager.closeFile(id);
+  }
+
+  /**
+   * Tests closing an unopened file fails.
+   */
+  @Test
+  public void closeNonExistentUfsFileTest() throws Exception {
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    mThrown.expect(FileDoesNotExistException.class);
+    manager.closeFile(-1L);
+  }
+
+  /**
+   * Tests getting an output stream for a valid file returns the correct output stream.
+   */
+  @Test
+  public void getOutputStreamTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    long id = manager.createFile(new AlluxioURI(uniqPath));
+    Assert.assertEquals(mMockOutputStream, manager.getOutputStream(id));
+  }
+
+  /**
+   * Tests getting an output stream from an invalid file fails.
+   */
+  @Test
+  public void getNonExistentOutputStreamTest() throws Exception {
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    mThrown.expect(FileDoesNotExistException.class);
+    manager.getOutputStream(-1L);
+  }
+
+  /**
+   * Tests getting an input stream to a valid file at the start returns the correct input stream.
+   */
+  @Test
+  public void getInputStreamTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    long position = 0L;
+    Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(true);
+    Mockito.when(mMockInputStream.skip(position)).thenReturn(position);
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    long id = manager.openFile(new AlluxioURI(uniqPath));
+    Assert.assertEquals(mMockInputStream, manager.getInputStreamAtPosition(id, position));
+    Mockito.verify(mMockInputStream).skip(position);
```

Comment:
Yes, but I don't think it matters for the mock stream, would you prefer to close it just to be a better example?